Updating the Collection, adding the new Release object (#126)
CycloneDX/transparency-exchange-api: tea-collection/tea-collection.md

# TEA release and collections

## The TEA release object (TRO)

> The TEA release object is a list of releases (versions) of a component

Reviewer: ```suggestion
The TEA Component Release object is a list of releases (versions) of a component
```

Might I suggest using "Component Release" instead of release. Consumers might have some idea of what a "Product Release" is and we should make clear that this is not a release of the TEA Product, but a release of the TEA Component.

  dynamically.
* __Static__: The TCO is built at publication time as a static
  object by the publisher. This object can be digitally signed at
  publication time and version controlled.

### Collection object

The TEA Collection object has the following parts

* Preamble

>   * UUID of the TEA collection object (TCO)

Reviewer: ```suggestion
  * UUID of the TEA component release object (TRO)
```

TEA Component Release and TEA Collection share the UUID. One of them should be the "owner" of the UUID.

>   * Product Release date (timestamp)

Reviewer: ```suggestion
  * Collection last modification date (timestamp)
```

Author: We may want both a collection last modification and original release date. The release date is critical for sorting. A collection may get updates after release.

Author: Moved to #134

Reviewer: In that case this field should be rather called `Component Release date` instead of `Product Release date` to avoid the confusions I mentioned in #131.

  * Version of this collection object
  * Reason for update/release of TCO
    * ENUM reason

>       See OpenAPI spec

Reviewer: In the OpenAPI spec we can give the values of the enum, but we can not give the descriptions.

We should point readers to the section below.

Author: Or just remove that comment

  * Size in bytes
  * SHA384 checksum

### The reason for TCO update enum

| ENUM        | Explanation                    |
|-------------|--------------------------------|

> | VEXUPDATE   | Updated the VEX artifact(s)    |
> | UPDATE  | Updated the artifact(s) other than VEX |
> | REMOVAL | Removal of artifact       |
> | ADDITION | Addition of an artifact |

Reviewer: ```suggestion
| INITIAL_RELEASE | Initial release of the collection |
| VEX_UPDATED   | Updated the VEX artifact(s)    |
| ARTIFACT_UPDATED  | Updated the artifact(s) other than VEX |
| ARTIFACT_REMOVED | Removal of artifact       |
| ARTIFACT_ADDED | Addition of an artifact |
```

These are the values in the OpenAPI Spec after #127 is merged.
